homeassistant/components/binary_sensor/ring.py

```diff
@@ -0,0 +1,135 @@
+"""
+This component provides HA sensor support for Ring Door Bell/Chimes.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/binary_sensor.ring/
+"""
+import logging
+from datetime import timedelta
+
+import voluptuous as vol
+import homeassistant.helpers.config_validation as cv
+import homeassistant.loader as loader
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorDevice, PLATFORM_SCHEMA)
+from homeassistant.const import (
+    CONF_ENTITY_NAMESPACE, CONF_MONITORED_CONDITIONS,
+    CONF_USERNAME, CONF_PASSWORD, ATTR_ATTRIBUTION)
+
+from requests.exceptions import HTTPError, ConnectTimeout
+
+REQUIREMENTS = ['ring_doorbell==0.1.1']
+
+_LOGGER = logging.getLogger(__name__)
+
+NOTIFICATION_ID = 'ring_notification'
+NOTIFICATION_TITLE = 'Ring Binary Sensor Setup'
+
+DEFAULT_CACHEDB = 'ring_cache.pickle'
+DEFAULT_ENTITY_NAMESPACE = 'ring'
+SCAN_INTERVAL = timedelta(seconds=5)
+
+CONF_ATTRIBUTION = "Data provided by Ring.com"
+
+# Sensor types: Name, category, device_class
+SENSOR_TYPES = {
+    'ding': ['Ding', ['doorbell'], 'occupancy'],
+    'motion': ['Motion', ['doorbell'], 'motion'],
+}
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_USERNAME): cv.string,
```

Comment:
Instead of having each platform require credentials, please add a component to manage the credentials and share the `Ring` instance by adding it to `hass.data` (a dictionary intended for sharing exactly such objects)